operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/AbstractResourceEventSource.java

```diff
@@ -15,4 +22,18 @@ protected AbstractResourceEventSource(Class<R> resourceClass) {
   public Class<R> resourceType() {
     return resourceClass;
   }
+
+  public void setOnAddFilter(Predicate<R> onAddFilter) {
```

Comment:
I don't think that the set methods should be exposed.

Reply:
This is for every event source, so also the external resources ones, will check, maybe to hide here and make it visible there.